sources/packages/web/src/components/common/notes/CreateNoteModal.vue

```diff
@@ -49,52 +71,66 @@ export default {
       required: true,
     },
   },
-  emits: ["submitData"],
-  setup(props: any, context: any) {
-    const { showDialog, showModal } = useModalDialog<void>();
-    const formData = ref();
-    const formioUtils = useFormioUtils();
-    const dialogClosed = () => {
-      showDialog.value = false;
-    };
-    const formLoaded = async (form: any) => {
-      formData.value = form;
-      const dropdown = formioUtils.getComponent(form, "noteType");
-      const options = [{}];
+  setup(props) {
+    const NOTES_MAX_CHARACTERS = 500;
+    const { checkMaxCharacters } = useValidators();
+    const { showDialog, showModal, resolvePromise } = useModalDialog<
+      NoteAPIInDTO | boolean
+    >();
+    const addNewNoteForm = ref({} as VForm);
+    const formModel = reactive({} as NoteAPIInDTO);
+    const noteTypeItems = ref([] as NoteTypeItems[]);
+
+    onMounted(async () => {
       if (props.entityType === NoteEntityType.Institution) {
         for (const noteType in InstitutionNoteType) {
-          options.push({ label: noteType, value: noteType });
+          noteTypeItems.value.push({ title: noteType, value: noteType });
```

Comment:
In general I would prefer replace the array at once like `noteTypeItems.value =` instead of `noteTypeItems.value.push`.
Since the array is reactive, does it means that vue will change the `Virtual DOM->DOM` every time that a new item is changed?